package-builder/flox-build.mk

```diff
@@ -157,6 +157,13 @@ define BUILD_local_template =
 	@echo "Building $(_name) in local mode"
 	$(if $(_virtualSandbox),$(PRELOAD_ARGS) FLOX_SRC_DIR=$$$$($(_pwd)) FLOX_VIRTUAL_SANDBOX=$(_sandbox)) \
 	MAKEFLAGS= out=$(_out) $(FLOX_ENV)/activate --turbo -- $(_bash) -e $($(_pvarname)_buildScript)
+	@if [ ! -e "$(_out)" ]; then \
+		echo "ERROR: Build command did not copy outputs to '\$$$$out'." 1>&2; \
+		echo "- copy a single file with 'cp bin \$$$$out'." 1>&2; \
+		echo "- copy multiple files with 'mkdir -p \$$$$out && cp bin/* \$$$$out/'" 1>&2; \
+		echo "- copy files from an Autotools project with 'make install PREFIX=\$$$$out'" 1>&2; \
+		exit 1; \
+	fi
```

Comment:
can this be handled in nix as well?

https://github.com/flox/flox/pull/2113/files#diff-7334023a7d623f0f6e31d47f68c0ccadac020a620b34867c77dd7feb9b810dd1R51-R54

Reply:
As in produce and test the same error when `buildScript == null` for pure builds?

Reply:
as invoke nix even if we know that $out is empty and handle that in nix

but yeah considering that we might want to _catch nix errors_ in make causing nix errors deliberately seems suboptimal...

Reply:
I can get close with this, which works for pure and impure:
```nix
assert pkgs.lib.assertMsg (builtins.pathExists install-prefix) ''
ERROR: Build command did not copy outputs to '$out'.
- copy a single file with 'cp bin $out'
- copy multiple files with 'mkdir -p $out && cp bin/* $out/'
- copy files from an Autotools project with 'make install PREFIX=$out'
'';
```

But I can't suppress the trace from the assertion itself which clutters the error message:
```
error:
       … in the condition of the assert statement

         at /nix/store/nk0f6n688y296aj7pb4hmrb4cimlmyh8-package-builder-1.0.0/libexec/build-manifest.nix:17:1:

           16| # out/install-prefix must exist
           17| assert pkgs.lib.assertMsg (builtins.pathExists install-prefix) ''
             | ^
           18| ERROR: Build command did not copy outputs to '$out'.

       … in the right operand of the OR (||) operator

         at /nix/store/72vz3idq9zwhmsbc0xssnyq52iziwlgy-source/lib/asserts.nix:42:10:

           41|     msg:
           42|     pred || builtins.throw msg;
             |          ^
           43|

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: ERROR: Build command did not copy outputs to '$out'.
       - copy a single file with 'cp bin $out'
       - copy multiple files with 'mkdir -p $out && cp bin/* $out/'
       - copy files from an Autotools project with 'make install PREFIX=$out'
```

Reply:
We paired on moving both checks into the build manifest. We still have to check twice, but at least they're confined to the same file and the output is the same for both pure and impure. I've rebased the changes into the existing commits and updated the PR description.